test/test.xunit.execution/Sdk/Frameworks/Runners/XunitTestCollectionRunnerTests.cs

```diff
@@ -38,6 +38,58 @@ public class XunitTestCollectionRunnerTests
         Assert.True(fixtureUnderTest.Disposed);
     }
 
+    [Fact]
+    public static async void DisposeAndAsyncLifetimeShouldBeCalledInTheRightOrder()
+    {
+        var collection = new TestCollection(Mocks.TestAssembly(), Reflector.Wrap(typeof(CollectionForFixtureAsyncLifetimeAndDisposableUnderTest)), null);
+        var testCase = Mocks.XunitTestCase<XunitTestCollectionRunnerTests>("DisposeAndAsyncLifetimeShouldBeCalledInTheRightOrder", collection);
+        var runner = TestableXunitTestCollectionRunner.Create(testCase);
+
+        var runnerSessionTask = runner.RunAsync();
+
+        await Task.Delay(500);
```

Comment:
Can't wait it any other way here, since I don't control the lifecycle of the fixture.

Reply:
:+1: